### PR TITLE
EB-1200: Implement signals in live preview sdk

### PR DESCRIPTION
**Pull request by Vishvam10**
### Jira Story

Work has been completed on this Jira story : https://contentstack.atlassian.net/browse/EB-1200

### Overview

This PR focuses on integrating `DeepSignal` package into existing codebase and refactoring the `Config` object to make it reactive. This will make things easier in the future (say if we want to subscribe to a property, etc) and also improves the DX.

### Changes Made

- Installed and configured `DeepSignal` package, a re-implementation of `Preact Signals` using `JS Proxies`
- Updated all the static methods of the `Config` object
- Refactored **all** Config related code to incorporate `DeepSignal`
- Re-wrote and corrected broken test cases related to the `Config` object

### CODE REVIEW

File: src/configManager/__test__/handleUserConfig.test.ts
Comment (Deepak-Kharah) [inline src/configManager/__test__/handleUserConfig.test.ts:1]: Test cases are missing

File: src/livePreview/editButton/editButton.ts [hunk 1]
```diff
@@ -311,8 +308,8 @@ export class LivePreviewEditButton {
 
         if (currentRectOfElement && currentRectOfParentOfElement) {
             const editButtonPosition = getEditButtonPosition(
-                elements.highlightedElement,
-                config.editButton.position
+                elements.highlightedElement as HTMLElement,
+                editButton.position as unknown as string
```
Comment (Deepak-Kharah) [inline src/livePreview/editButton/editButton.ts:312]: Let's remove all the as unknown code

**Review by Deepak-Kharah — CHANGES_REQUESTED** [review]

File: src/livePreview/eventManager/postMessageEvent.hooks.ts [hunk 2]
```diff
@@ -46,29 +46,24 @@ export function useOnEntryUpdatePostMessageEvent(): void {
...
             setConfigFromParams({
                 live_preview: event.data.hash,
             });
-
-            if (!config.ssr) {
-                const config = Config.get();
-                config.onChange();
+            const { ssr, onChange } = Config.get();
+            if (!ssr) {
+                onChange!();
```
[the remark this answers is not in the document]
Comment (Vishvam10) [reply src/livePreview/eventManager/postMessageEvent.hooks.ts:54]: Hmm. Not really. I'll remove it

File: src/livePreview/editButton/editButton.ts [hunk 3]
```diff
@@ -190,16 +192,17 @@ export function shouldRenderEditButton(editButton: IConfigEditButton): boolean {
         PublicLogger.error(error);
     }
 
-    // case if inside live preview
+    console.log('[IN SDK] : WINDOWTYPE : ', Config.get().windowType);
```
Comment (Deepak-Kharah) [inline src/livePreview/editButton/editButton.ts:195]: remove the logs

**Review by Deepak-Kharah — CHANGES_REQUESTED** [review]

**Review by Deepak-Kharah — APPROVED** [review]

**Pull request merged**